acre/lib.py

```diff
@@ -37,12 +38,24 @@ class FormatDict(dict):
         Missing keys are replaced with the return value of __missing__.
 
         """
+
         def __missing__(self, key):
             return missing.format(key=key)
 
     formatter = string.Formatter()
     mapping = FormatDict(**data)
-    return formatter.vformat(s, (), mapping)
+    try:
+        f = formatter.vformat(s, (), mapping)
+    except Exception:
+        r_token = re.compile(r"({.*?})")
+        matches = re.findall(r_token, s)
+        f = s
+        for m in matches:
+            try:
+                f = re.sub(m, m.format(**data), f)
+            except KeyError:
+                continue
+    return f
```

Comment:
Could we improve the Exception (e.g. more explicit; what exception should we capture?) and add a comment to the code as to what this is trying to solve? When should this exception occur?